From 2355af60dc0c0ec2b7fbe69f15a77d980b017b3f Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Sat, 4 Oct 2014 13:10:41 +0200
Subject: [PATCH] consoled: add a unit file

The unit file is statically enabled, but still requires --enable-terminal
to actually get installed.
---
 Makefile.am                            | 11 +++++++++++
 units/user/.gitignore                  |  1 +
 units/user/systemd-consoled.service.in | 15 +++++++++++++++
 3 files changed, 27 insertions(+)
 create mode 100644 units/user/systemd-consoled.service.in

diff --git a/Makefile.am b/Makefile.am
index 3119a95dad..e52db1793b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -250,6 +250,7 @@ SOCKETS_TARGET_WANTS =
 BUSNAMES_TARGET_WANTS =
 TIMERS_TARGET_WANTS =
 USER_SOCKETS_TARGET_WANTS =
+USER_DEFAULT_TARGET_WANTS =
 USER_BUSNAMES_TARGET_WANTS =
 
 SYSTEM_UNIT_ALIASES =
@@ -270,6 +271,7 @@ install-target-wants-hook:
 	what="$(TIMERS_TARGET_WANTS)" && wants=timers.target && dir=$(systemunitdir) && $(add-wants)
 	what="$(SLICES_TARGET_WANTS)" && wants=slices.target && dir=$(systemunitdir) && $(add-wants)
 	what="$(USER_SOCKETS_TARGET_WANTS)" && wants=sockets.target && dir=$(userunitdir) && $(add-wants)
+	what="$(USER_DEFAULT_TARGET_WANTS)" && wants=default.target && dir=$(userunitdir) && $(add-wants)
 
 install-busnames-target-wants-hook:
 	what="$(BUSNAMES_TARGET_WANTS)" && wants=busnames.target && dir=$(systemunitdir) && $(add-wants)
@@ -3035,6 +3037,15 @@ noinst_PROGRAMS += \
 dist_pkgdata_DATA += \
 	src/libsystemd-terminal/unifont-glyph-array.bin
 
+nodist_userunit_DATA += \
+	units/user/systemd-consoled.service
+
+USER_DEFAULT_TARGET_WANTS += \
+	systemd-consoled.service
+
+EXTRA_DIST += \
+	units/user/systemd-consoled.service.in
+
 tests += \
 	test-term-page \
 	test-term-parser \
diff --git a/units/user/.gitignore b/units/user/.gitignore
index 6aa54754cb..c91ed626c8 100644
--- a/units/user/.gitignore
+++ b/units/user/.gitignore
@@ -1,2 +1,3 @@
 /systemd-exit.service
 /systemd-bus-proxyd@.service
+/systemd-consoled.service
diff --git a/units/user/systemd-consoled.service.in b/units/user/systemd-consoled.service.in
new file mode 100644
index 0000000000..fd7938aa8b
--- /dev/null
+++ b/units/user/systemd-consoled.service.in
@@ -0,0 +1,15 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Console Manager and Terminal Emulator
+
+[Service]
+Type=notify
+Restart=always
+RestartSec=0
+ExecStart=@rootlibexecdir@/systemd-consoled
